pkg/storage/internal/repo/outputs.go

```diff
@@ -96,3 +97,50 @@ func (o *Outputs) mapModelToTableOutput(model *models.Output) *wdk.TableOutput {
 	}
 	return output
 }
+
+func listAndCountOutputs(ctx context.Context, db *gorm.DB, params entity.ListOutputsParams) ([]*models.Output, int64, error) {
+	var outputs []*models.Output
+	var total int64
+
+	if err := db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
+		query := tx.Model(&models.Output{}).Where("user_id = ?", params.UserID)
+
+		if params.BasketName != "" {
+			query = query.Where("basket_id IN (?)", db.Model(&models.OutputBasket{}).Select("basket_id").Where("name = ?", params.BasketName))
```

Comment:
There could be several rows with the same name - but we are interested in provided-user baskets.
The output is the same (I believe), but there are less "unnecessary" IDs provided to `IN` clause. 

```suggestion
			query = query.Where("basket_id IN (?)", db.Model(&models.OutputBasket{}).Select("basket_id").Where("name = ? and user_id = ?", params.BasketName, params.UserID))
```